index.bs

```diff
@@ -266,14 +269,51 @@ Privacy & Security Considerations {#privacy-security}
 
 The requirement that the provided camera images are [=aligned=] with the {{XRView}}s offers a partial mitigation for impact of the API on users' privacy. A consequence of this requirement is that depending on the [=XR Compositor=]'s [=blend technique=], the camera image will contain the same information that the user may already be seeing, thus providing direct feedback about which parts of the environment are visible to the website. Note that in some cases it would be possible for the sites to suppress this information (e.g. on smartphones by rendering an opaque object that would encompass the entire view, thus occluding the environment rendered by the compositor).
 
 In addition to the requirement for the user agents to seek user consent, the application developers are strongly encouraged not to ask for [=camera-access=] feature if there are other means of achieving their use cases.
 
 </section>
 
+User experience {#user-experience}
+---------------
+
+<section class="non-normative">
+
+As stated above, due to privacy implications of Raw Camera Access API, the user agent should seek user consent prior to creating a session with [=camera-access=] feature enabled. This can be achieved in multiple ways, some of which would require interacting with the user directly.
+
+Some examples of interacting with the user directly are:
+1. Displaying a permission prompt that allows the user to select how fine-grained information will be exposed to the site. Mock of such prompt can be seen below.
+
+<p align="center">
+  <img src="img/permission-mock.png"></img>
+</p>
+
+2. Displaying a series of permission prompts that would take into consideration various levels of user consent needed in order to create a session with the requested and optional features provided by the application to {{XRSystem/requestSession()}} call.
+
+</section>
+
+Note: It is the intent of this specification to ensure that the user agent does not encourage app developers to ask for [=camera-access=] feature if the experience they want to provide does not absolutely require it. Due to privacy implications of the feature, user agents are allowed to introduce additional friction when a site asks for access to the camera images, with the hopes that this would incentivize the applications to not ask for the feature needlessly.
+
+Privacy indicators {#privacy-indicators}
+------------------
+
+The user agent SHOULD attempt to display a privacy indicator every time an {{XRSession}} with [=camera-access=] has been created and has not yet [=XRSession/ended=]. If it ever becomes possible to modify a [=XRSession/set of granted features=] of a session after a session has been created, the indicator SHOULD be displayed for at least as long as the [=camera-access=] feature descriptor is [=list/contain|contained=] in the [=XRSession/set of granted features=] of the session.
```

Comment:
"SHOULD attempt to" probably is more spec-texty if it is simplified to "SHOULD" 

That said, any reason why this is not a MUST?

Reply:
Sounds good, I'll switch it to MUST - I was a bit worried about potential form factors where showing the indicator is impossible, but I think we can tighten it now and think about relaxing it if need arises.